src/types/query.rs

```diff
@@ -0,0 +1,64 @@
+use edn_rs::Serialize;
+
+#[derive(Clone)]
+pub struct Query {
+    find: Vec<String>,
+    whr: Option<Vec<String>>,
+}
+
+impl Query {
+    pub fn find(find: Vec<&str>) -> Self {
+        Self {
+            find: find
+                .into_iter()
+                .map(|f| String::from(f))
+                .collect::<Vec<String>>(),
+            whr: None,
+        }
+    }
+
+    pub fn where_clause(mut self, whr: Vec<&str>) -> Self {
+        self.whr = Some(
+            whr.into_iter()
+                .map(|f| String::from(f))
+                .collect::<Vec<String>>(),
```

Comment:
Same thing of above about the necessity of type notations

Reply:
```
expected struct `std::vec::Vec<std::string::String>`
   found struct `std::vec::Vec<&str>`rustc(E0308)
```